PyMca5/PyMcaGui/io/TiledCatalogSelector.py

```diff
@@ -0,0 +1,164 @@
+import logging
+from collections import defaultdict
+from typing import Callable, List, Mapping, Optional, Sequence
+from urllib.parse import ParseResult, urlparse as _urlparse
+
+from PyQt5.QtCore import QEvent, QObject, pyqtSignal
+
+from tiled.client import from_uri
+from tiled.client.base import BaseClient
+
+
+_logger = logging.getLogger(__name__)
+
+
+class TiledCatalogSelectorSignals(QObject):
+    """Collection of signals for a TiledCatalogSelector model."""
+    client_connected = pyqtSignal(
+        str, # new URL
+        str, # new API URL
+        name="TiledCatalogSelector.client_connected",
+    )
+    client_connection_error = pyqtSignal(
+        str, # Error message
+        name="TiledCatalogSelector.client_connection_error",
+    )
+    url_validation_error = pyqtSignal(
+        str, # Error message
+        name="TiledCatalogSelector.url_validation_error",
+    )
+    
+    def __init__(self, parent: Optional[QObject] = None) -> None:
+        super().__init__(parent)
+
+
+class TiledCatalogSelector(object):
+    """View Model for selecting a Tiled CatalogOfBlueskyRuns."""
+    Signals = TiledCatalogSelectorSignals
+
+    def __init__(
+        self,
+        /,
+        url: str = None,
+        client: BaseClient = None,
+        validators: Mapping[str, List[Callable]] = None,
+        parent: Optional[QObject] = None,
+        *args,
+        **kwargs,
+    ):
+        _logger.debug("TiledCatalogSelector.__init__()...")
+
+        self.url = url
+        self.client = client
+        self.validators = defaultdict(list)
+        if validators:
+            self.validators.update(validators)
+
+        self.signals = self.Signals(parent)
+        self.client_connected = self.signals.client_connected
+        self.client_connection_error = self.signals.client_connection_error
+        self.url_validation_error = self.signals.url_validation_error
+
+        # A buffer to receive updates while the URL is being edited
+        self.url_buffer = None
```

Comment:
I wonder what the trade-offs are between using a type-stable `""` versus an explicit `None`.

Reply:
I think I originally had a reason for wanting to distinguish between an empty string supplied by  a user vs nothing supplied by a user.  But there may be no difference in practice.

I suppose as a test, we could switch to an empty string until we find a practical reason to employ some sentinel, like `None`.

Reply:
Updated to use `""` in commit `97176bcf`.

Reply:
Did this change get pushed? I don't see this commit.